### PR TITLE
[#9795] prevent duplicate mcq options

### LINKED ISSUE #9795: MCQ: Prevent duplicate options in MCQ
- **Environment**: 
`master` branch at commit 30f56ffea3016decff6026cc46bb03e346ab7ffd

**UPDATE**
I realised the options I set are the same and that caused the problem. Perhaps the issue should be "Prevent duplicate options in MCQ" instead.

**Steps to reproduce**
~~1. Go to an unanswered MCQ question in a session~~
~~2. Select any option besides the first option~~

**Expected behaviour**
~~Option you select is selected~~

**Actual behaviour**
~~The first option will be selected~~

### PR DESCRIPTION
**Pull request by nusjzx**
Fixes #9795 

**Outline of Solution**

how to check duplicate: (number of distinct elements == number of total elements)

### CODE REVIEW

File: src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java [hunk 1]
```diff
@@ -605,6 +605,12 @@ public String getQuestionTypeChoiceOption() {
...
                 }
             }
+
+            //If there are duplicate mcq options trigger this error
+            boolean isDuplicateOptionsEntered = mcqChoices.stream().map(String::trim).distinct().count() != mcqChoices.size();
+            if(isDuplicateOptionsEntered) {
+                errors.add(Const.FeedbackQuestion.MCQ_ERROR_DUPLICATE_MCQ_OPTION);
+            }
         }
         //TODO: check that mcq options do not repeat. needed?
```
Comment (xpdavid) [inline src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java]: You should delete the TODO.

File: src/test/java/teammates/test/cases/datatransfer/FeedbackMcqQuestionDetailsTest.java [hunk 2]
```diff
@@ -385,6 +385,33 @@ public void testValidateQuestionDetails_negativeOtherWeight_errorsReturned() {
         assertEquals(Const.FeedbackQuestion.MCQ_ERROR_INVALID_WEIGHT, errors.get(0));
     }
 
+    @Test
+    public void testValidateQuestionDetails_duplicateMcqOptions_errorReturned() {
```
Comment (xpdavid) [inline src/test/java/teammates/test/cases/datatransfer/FeedbackMcqQuestionDetailsTest.java]: BTW, you should test directly the method `validateQuestionDetails`.

`extractQuestionDetails` is deprecated in V7 now.

File: src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java (hunk 1 above)
Comment (nusjzx) [reply src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java]: Sure

File: src/test/java/teammates/test/cases/datatransfer/FeedbackMcqQuestionDetailsTest.java (hunk 2 above)
Comment (nusjzx) [reply src/test/java/teammates/test/cases/datatransfer/FeedbackMcqQuestionDetailsTest.java]: done

File: src/test/java/teammates/test/cases/datatransfer/FeedbackMsqQuestionDetailsTest.java [hunk 3]
```diff
@@ -389,6 +389,34 @@ public void testValidateQuestionDetails_negativeOtherWeight_errorsReturned() {
         assertEquals(Const.FeedbackQuestion.MSQ_ERROR_INVALID_WEIGHT, errors.get(0));
     }
 
+    @Test
+    public void testValidateQuestionDetails_duplicateMsqOptions_errorReturned() {
```
[the remark this answers is not in the document]
Comment (nusjzx) [reply src/test/java/teammates/test/cases/datatransfer/FeedbackMsqQuestionDetailsTest.java]: Sure

**Pull request merged**